### DISCUSSION TITLE
Best practices for zoom-dependent styling?

### DISCUSSION THREAD

**Discussion by tylere** [developmentseed/lonboard#616] [Q&A]
What are the options / best practices for altering the styling of layers based on the zoom level?

One option could be to observe map widget event changes (similar to the following example), but is there a better approach?

```python
m = lonboard.Map(layers)

def on_map_change(change):
    if change['name'] == '_property_lock':
        try:
            zoom = change['new']['view_state']['zoom']
            # code to manipulate layers
            layers[1].visible = (zoom < 1)
        except KeyError:
            pass   
m.observe(on_map_change)
```

**Comment by kylebarron** [comment]
The easiest way is when the upstream deck.gl properties automatically support zoom-based styling. That is, for radius, there's `radiusUnits` that you can set to `meters`, which will automatically scale the radius at different zooms for roughly the same physical on-earth size.

Otherwise, I don't have great advice for the general case. I think what you have there is a good start!
